dotnetv3/dynamodb/mid-level-api/MidLevelQueryAndScanExample/MidLevelQueryAndScanExample/MidLevelQueryAndScan.cs

```diff
@@ -0,0 +1,219 @@
+            filter.AddCondition("ReplyDateTime", QueryOperator.GreaterThan, twoWeeksAgoDate);
+
+            // Use Query overloads that takes the minimum required query parameters.
+            Search search = table.Query(filter);
+
+            do
+            {
+                var documentSet = await search.GetNextSetAsync();
+                Console.WriteLine("\nFindRepliesInLast15Days: printing ............");
+
+                foreach (var document in documentSet)
+                {
+                    PrintDocument(document);
+                }
+            }
+            while (!search.IsDone);
+        }
+
+        /// <summary>
+        /// Retrieve replies made during a specific time period.
+        /// </summary>
+        /// <param name="table">The table we want to query.</param>
+        /// <param name="forumName">The name of the forum we are interested in.</param>
+        /// <param name="threadSubject">The subject of the thread which we are
+        /// searching for replies.</param>
+        public static async Task FindRepliesPostedWithinTimePeriod(
+          Table table,
+          string forumName,
+          string threadSubject)
+        {
+            DateTime startDate = DateTime.UtcNow.Subtract(new TimeSpan(21, 0, 0, 0));
+            DateTime endDate = DateTime.UtcNow.Subtract(new TimeSpan(1, 0, 0, 0));
+
+            QueryFilter filter = new QueryFilter("Id", QueryOperator.Equal, forumName + "#" + threadSubject);
+            filter.AddCondition("ReplyDateTime", QueryOperator.Between, startDate, endDate);
+
+            QueryOperationConfig config = new QueryOperationConfig()
+            {
+                Limit = 2, // 2 items/page.
+                Select = SelectValues.SpecificAttributes,
+                AttributesToGet = new List<string>
+        {
+          "Message",
+          "ReplyDateTime",
+          "PostedBy",
+        },
+                ConsistentRead = true,
+                Filter = filter,
+            };
+
+            Search search = table.Query(config);
+
+            do
+            {
+                var documentList = await search.GetNextSetAsync();
+                Console.WriteLine("\nFindRepliesPostedWithinTimePeriod: printing replies posted within dates: {0} and {1} ............", startDate, endDate);
+
+                foreach (var document in documentList)
+                {
+                    PrintDocument(document);
+                }
+            }
+            while (!search.IsDone);
+        }
+
+        /// <summary>
+        /// Perform a query for replies made in the last 15 days using a DynamoDB
+        /// QueryOperationConfig object.
+        /// </summary>
+        /// <param name="table">The table we want to query.</param>
+        /// <param name="forumName">The name of the forum we are interested in.</param>
+        /// <param name="threadName">The bane of the thread which we are looking
```

Comment:
The name of the thread that we're searching

Reply:
fixed